File: trunk/ShadowFrameworkLite20/src/shadow/renderer/data/SFRendererData.java
package shadow.renderer.data;

import shadow.renderer.SFCamera;
import shadow.renderer.SFProgramModuleStructures;
import shadow.renderer.SFRenderer;
import shadow.system.data.SFNamedParametersObject;

public class SFRendererData extends SFDataAsset<SFRenderer>{

	private SFDataAssetObject<SFCamera> camera = new SFDataAssetObject<SFCamera>(null);
	protected SFDataAssetObject<SFProgramModuleStructures> light=new SFDataAssetObject<SFProgramModuleStructures>(new SFProgramAssetData());
	
	public SFRendererData() {
		prepare();
	}
	
	public SFRendererData(String lightProgramName,SFDataAsset<SFCamera> camera) {
		prepare();
		this.camera.setDataset(camera);
		((SFProgramAssetData)this.light.getDataset()).setProgram(lightProgramName);
	}
	

	private void prepare() {
		SFNamedParametersObject parameters=new SFNamedParametersObject();
		parameters.addObject("camera", camera);
		parameters.addObject("light", light);
		setData(parameters);
	}
	
	@Override
	protected SFRenderer buildResource() {
		final SFRenderer renderer = new SFRenderer();
		renderer.setLight(light.getResource());
		renderer.setCamera(this.camera.getDataset().getResource());
		return renderer;
	}

}

File: ShadowFrameworkJS/src/shadow/renderer/data/SFRendererData.js


function SFRendererData(){

	this.camera = new SFDataAssetObject(null);
	this.light=new SFDataAssetObject(new SFProgramAssetData());
	
	this.prepare();			
}

inherit(SFRendererData,SFDataAsset);


SFRendererData.prototype["prepare"]=function(){
	var parameters=new SFNamedParametersObject();
	parameters.addObject(this.camera);
	parameters.addObject(this.light);
	this.setData(parameters);
};


SFRendererData.prototype["buildResource"]=function(){
	var renderer = new SFRenderer();
	renderer.setLight(this.light.getResource());
	renderer.setCamera(this.camera.getDataset().getResource());
	return renderer;
};
	

SFRendererData.prototype["generateNewDatasetInstance"]=function(){
	return new SFRendererData();
};
